### DISCUSSION TITLE
Cannot set recording directory in version 3.8.1 on Mac

### DISCUSSION THREAD

**Discussion by Rob-NY** [jamulussoftware/jamulus#2251] [Bug Investigation]
Running the most recent update to Big Sur on a MBP.  Downloaded fresh 3.8.1 dmg and installed it.  When trying to set the Recording Directory, not only does it not properly browse the file system (directories that really exist do not appear, and directories that shouldn't be there do appear), but when choosing ANY directory it says "not writable".  For example, I cannot even navigate to my documents, music, or downloads directories.  I can get sort of get to /Users/myname and then everything gets weird.  Something, perhaps, with the QT File dialog?

Uninstalled 3.8.1 completely and installed 3.8.0 from releases section of github.  No issues.  All works as normal.

I then tested the 3.8.1 LEGACY dmg and it worked fine even though I'm on Big Sur.   

Not sure if this is a documentation issue surrounding which mac os versions should be using "legacy" or if it is really a bug in 3.8.1.

Let me know if you need more info or if I screwed something up.

**Comment by hoffie** [comment]
This sounds like it might be related to the signing/sandboxing changes. A similar file access issue regarding config files is being investigated in #2131. Might be worth following.

cc @emlynmac

**Comment by Mumajor** [comment]
I've had the same issue with setting the recording directory for any server version past 3.8.0. I'm running 3.8.0 on a Mac M1 (2020) Mini with Big Sur 11.7.3 and all versions of clients up to 3.9.1 appear to work well with server 3.8.0, however no newer server version allows the recording directory to be set.

The common message is below

ERROR: '/......./Jamulus' is a directory but cannot be written to

Is there a known fix for this bug? Any insight is much appreciated!

**Reply by pljones** [reply]
Have you taken a look at #2131, linked above?

**Reply by Mumajor** [reply]
Thanks for the suggestion, I appreciate the response.

I have looked at that link and discussion but my skill set is this area is completely lacking. Luckily 3.8.0 server continues to work for me, so it looks like I'll have to stay at that revision.

**Reply by ann0see** [reply]
Ok. As this came up again, It might be worth re investigating the problem
